service/grails-app/services/org/olf/licenses/LicenseHousekeepingService.groovy

```diff
@@ -0,0 +1,38 @@
+package org.olf.licenses
+
+import grails.gorm.transactions.Transactional
+import org.olf.licenses.License
+import org.olf.licenses.LicenseAmendment
+import com.k_int.web.toolkit.refdata.RefdataValue
+
+/**
+ * This service works at the module level, it's often called without a tenant context.
+ */
+@Transactional
+public class LicenseHousekeepingService {
+
+  public void triggerHousekeeping() {
+    this.checkUnsetValues();
+  }
+
+
+  def checkUnsetValues() {
+        log.debug("EndDateSemanticsCleanup: Check for unset values")
+
+          int i = 0
+          License.findAllByEndDateSemanticsIsNull().each {lic ->
```

Comment:
If we really need to display this count in the log message I'd recommend either saving `License.findAllByEndDateSemanticsIsNull` to a variable and counting before running through with `.each`, or utilising something like Grails' `countBy`: http://docs.grails.org/3.1.1/ref/Domain%20Classes/countBy.html


However I'm not entirely sure we actually need this kind of counter in the log messages.